doc/how-to-analyze-haskell-code-performance.rst

```diff
@@ -0,0 +1,164 @@
+How to analyze Haskell performance
+==================================
+
+When a Haskell application is slow or uses too much memory,
+Cabal and `GHC <https://downloads.haskell.org/ghc/latest/docs/users_guide/profiling.html>`__
+can help you understand why.
+
+The main steps are to
+
+1. let GHC insert performance measuring code into your application (or insert them
+   `manually <https://downloads.haskell.org/ghc/latest/docs/users_guide/profiling.html#inserting-cost-centres-by-hand>`__),
+2. run the application with the right
+   `runtime system (RTS) flags <https://downloads.haskell.org/ghc/latest/docs/users_guide/runtime_control.html>`__
+   to produce a performance report and
+3. visualize and analyze that report.
```

Comment:
These three steps, ideally, should serve as an outline to what follows. So, I'd suggest to rephrase the first one a little bit to mention "configuring", and the second one will map ideally on `cabal run` once you introduce it.

Also, the `N.`-lists should use sentence case (capital letters at the start of every point).

```suggestion
1. Configure the project in a way that makes GHC insert performance-measuring code into your application (or insert them.
2. Run the application with the right
   `runtime system (RTS) flags <https://downloads.haskell.org/ghc/latest/docs/users_guide/runtime_control.html>`__.
   to produce a performance report and
3. Visualize and analyze that report.
```

And what follows below should mention "step one", "step two", "step three", hopefully.

One more thing: I don't think mentioning manual insertion of cost centers should be done here. It's a detail that I'd save for a later time. Hence, I remove it from here.